contracts/LimitsChecker.sol

```diff
@@ -0,0 +1,234 @@
+    // EVENTS
+    // -------------
+    event LimitsParametersChanged(uint256 _limit, uint256 _periodDurationMonth);
+    event FundsSpent(
+        uint256 _alreadySpentAmount,
+        uint256 _spendableAmount,
+        uint256 _periodStartTimestamp,
+        uint256 _periodEndTimestamp
+    );
+
+    // -------------
+    // ERRORS
+    // -------------
+    string private constant ERROR_WRONG_PERIOD_DURATION = "WRONG_PERIOD_DURATION";
+    string private constant ERROR_SUM_EXCEEDS_LIMIT = "SUM_EXCEEDS_LIMIT";
+    string private constant ERROR_CALLER_IS_FORBIDDEN = "CALLER_IS_FORBIDDEN";
+    // -------------
+    // ROLES
+    // -------------
+    bytes32 public constant SET_LIMIT_PARAMETERS_ROLE = keccak256("SET_LIMIT_PARAMETERS_ROLE");
+
+    // -------------
+    // CONSTANTS
+    // -------------
+
+    // ------------
+    // STORAGE VARIABLES
+    // ------------
+
+    /// @notice Address of BokkyPooBahsDateTimeContract
+    IBokkyPooBahsDateTimeContract public immutable bokkyPooBahsDateTimeContract;
+
+    /// @notice Address of EasyTrack
+    EasyTrack public immutable easyTrack;
+
+    /// @notice Length of period in months
+    uint256 internal periodDurationMonth;
+
+    /// @notice End of the current period
+    uint256 internal currentPeriodEnd;
+
+    /// @notice The maximum that can be spent in a period
+    uint256 internal limit;
+
+    /// @notice Amount already spent in the period
+    uint256 internal spent;
+
+    // ------------
+    // CONSTRUCTOR
+    // ------------
+    constructor(
+        EasyTrack _easy_track,
+        address[] memory _setLimitParametersRoleHolders,
+        IBokkyPooBahsDateTimeContract _bokkyPooBahsDateTimeContract
+    ) {
+        easyTrack = _easy_track;
+        for (uint256 i = 0; i < _setLimitParametersRoleHolders.length; i++) {
+            _setupRole(SET_LIMIT_PARAMETERS_ROLE, _setLimitParametersRoleHolders[i]);
+        }
+        bokkyPooBahsDateTimeContract = _bokkyPooBahsDateTimeContract;
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    /// @notice Checks if _payoutSum is less than may be spent in the period
+    /// @param _payoutSum Motion sum
+    /// @return True if _payoutSum is less than may be spent in the period
+    function isUnderSpendableBalance(uint256 _payoutSum) external view returns (bool) {
+        uint256 _motionDuration = easyTrack.motionDuration();
+        if (block.timestamp + _motionDuration >= currentPeriodEnd) {
+            return _payoutSum <= limit;
+        } else {
+            return _payoutSum <= limit - spent;
+        }
+    }
+
+    /// @notice Checks if _payoutSum is less than may be spent,
+    /// @notice updates period if needed and increases the amount spent in the current period
+    function updateSpendableBalance(uint256 _payoutSum) external {
+        require(msg.sender == address(easyTrack.evmScriptExecutor()), ERROR_CALLER_IS_FORBIDDEN);
+
+        _checkAndUpdateLimitParameters();
```

Comment:
Consider adding `assert(periodDurationMonth > 0);` for clarity before because otherwise the contract will revert with "modulo div by zero" inside `_getPeriodStartFromTimestamp` if `periodDurationMonth == 0`.

The same problem exists for calling `getCurrentPeriodState` when contract is in state when no call to `setLimitParameters` done.

Reply:
added check for zero periodDuration

Reply:
I see now: `updateSpendableBalance` reverts with `WRONG_PERIOD_DURATION`.

For `getCurrentPeriodState` the problem is still there, it reverts without proper reason due to division by zero.

I'd do:

```
function _getFirstMonthInPeriodFromCurrentMonth(...) {
  require(periodDurationMonth != 0, "ZERO_PERIOD_DURATION");
  ...
}
```

and update doc of `getCurrentPeriodState` that it reverts with `ZERO_PERIOD_DURATION` if no period is set.

Reply:
Thank you! made a check for periodDurationMonth != 0 but with INVALID_PERIOD_DURATION_ERROR.
Do we need separate error for zero check? wdyt?

Reply:
I think this check is enough: caller will get the error message and it's descriptive enough.